### PR TITLE
Move lint jobs to CircleCI

### PR DESCRIPTION
**Pull request by teohhanhui**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | N/A
| License       | MIT
| Doc PR        | N/A

### CODE REVIEW

**Comment by bendavies** [comment]
you can add a php-cs-fixer fixer cache

**Comment by teohhanhui** [comment]
@bendavies Good catch! :D

File: .circleci/config.yml [hunk 1]
```diff
@@ -45,6 +51,16 @@ reusable-steps:
       paths:
         - ~/.npm
       key: npm-cache-{{ .Revision }}-{{ .BuildNum }}
+  - &save-php-cs-fixer-cache-by-branch
+    save_cache:
+      paths:
+        - .php_cs.cache
+      key: php-cs-fixer-cache-{{ .Branch }}-{{ .BuildNum }}
```
Comment (bendavies) [inline .circleci/config.yml]: are you sure this is correct?
you never restore using `.BuildNum` so it seems pointless.
(i'm aware of how circle ci `restore_cache` works with substrings)

Comment (bendavies) [reply .circleci/config.yml]: i thought you might
save key: 
```
php-cs-fixer-cache-{{ .Branch }}-{{ .Revision }}
```
restore keys:
```
keys:
  - php-cs-fixer-cache-{{ .Branch }}-{{ .Revision }}
  - php-cs-fixer-cache-{{ .Branch }}
  - php-cs-fixer-cache
```

**Comment by bendavies** [comment]
there is npm caching in here, but i can't see that npm is ever ran. am i missing it?

File: .circleci/config.yml (hunk 1 above)
Comment (teohhanhui) [reply .circleci/config.yml]: It's intentional. The .BuildNum is for cache busting.

**Comment by teohhanhui** [comment]
@bendavies `npx` ;)

**Comment by bendavies** [comment]
ah! thanks

File: .circleci/config.yml (hunk 1 above)
Comment (bendavies) [reply .circleci/config.yml]: ah, in my project i actually generate a hash of all php files they php-cs-fixer runs over, then cache on that.

something like:
`find src tests -type f -name "*.php" -exec md5sum {} + > php-cs-fixer.checksum`

```
- restore_cache:
  keys:
    - php-cs-fixer-{{ checksum "php-cs-fixer.checksum" }}
    - php-cs-fixer-
```

then there is no need for cache busting.

**Comment by teohhanhui** [comment]
Overall build time should be back down to under 10 minutes (roughly). :tada:

**Comment by Simperfit** [comment]
Thanks @teohhanhui

**Pull request merged**